go/vt/sqlparser/ast.go

```diff
@@ -2284,7 +2300,7 @@ func (ct *ColumnType) DescribeType() string {
 
 // SQLType returns the sqltypes type code for the given column
 func (ct *ColumnType) SQLType() querypb.Type {
-	switch ct.Type {
+	switch strings.ToLower(ct.Type) {
```

Comment:
Nice catch! 

It would be good to retroactively add in some test case that would have been triggered by this, in order to cover this and prevent a future regression. Maybe just updating an existing test case to use a mIxEd case type name or something? Would have have been triggered by this?

Reply:
The only place using `.SQLType()` is here. I was confused on `INT` type because it was not giving detailed error on its type but then I just noticed it does 'fallthrough' for this type only. But, the current error test for `INT SRID 0` (I'll add an extra one with `FLOAT SRID 0` ) will cover this case and prevent regressions on this.